File: applications/Rearrange.java
/* Rearrange.java
 *
 */

import java.util.Arrays;

public class Rearrange {
  static public int[] rearrangePositiveNegative(int arr[]) {
    int j = 1;
    int i = 0;
    while(i < arr.length) {
      if((arr[i] > 0 && arr[j] < 0)) {
        i++;
        j++;
      } else if ((arr[i] < 0 && arr[j] < 0) || (arr[i] > 0 && arr[j] > 0)) {
        System.out.println(arr[i] + "  " + arr[j] + " " + Arrays.toString(arr) + "i" + i + "j" + j);
        j++;
        break;
      } else if (arr[i] < 0 && arr[j] > 0) {
        int temp = arr[i];
        arr[i] = arr[j];
        arr[j] = temp;
        System.out.println("i => "+ i+"j "+j);
      }
      j++;
      i++;
    }
    System.out.println(Arrays.toString(arr));
    return arr;
  }
  public static void main(String[] args) {
    int arr[] = {1, -1, -2, 5, 3, -4, -1, 4};
    System.out.println("Testing");
    Rearrange.rearrangePositiveNegative(arr);
  }
}

File: applications/Rearrange.js
/*
 *
 *
 *
 *
 */

function rearrange(arr) {
  let i = 0;
  let j = 1;
  while(arr.length > i) {
    if (arr[i] > 0 && arr[j] < 0) {
      console.log('Correct')
    } else {
      let temp = arr[i]
      arr[i] = arr[j]
      arr[j] = temp
      j--;
      console.log(arr)
    }
    i++;
    j++;
  }
}

rearrange([12, 11,-13,-5,6,-7,5,-3,-6])
